packages/core/src/data-editor/data-editor.tsx

```diff
@@ -2987,12 +3055,28 @@ const DataEditorImpl: React.ForwardRefRenderFunction<DataEditorRef, DataEditorPr
             if (gridSelection.current !== undefined && (movX !== 0 || movY !== 0)) {
                 const isEditingTrailingRow =
                     gridSelection.current.cell[1] === mangledRows - 1 && newValue !== undefined;
-                updateSelectedCell(
-                    clamp(gridSelection.current.cell[0] + movX, 0, mangledCols.length - 1),
-                    clamp(gridSelection.current.cell[1] + movY, 0, mangledRows - 1),
-                    isEditingTrailingRow,
-                    false
-                );
+                const isEditingTrailingCol =
+                    gridSelection.current.cell[0] === mangledCols.length - 1 && newValue !== undefined;
```

Comment:
nit: does it make sense to rename this to `isEditingLastCol` and `isEditingLastRow` to better differentiate from the `trailingRow` concept?

Reply:
Updated!